src/gui/socketapi.cpp

```diff
@@ -573,6 +605,19 @@ void SocketApi::command_GET_MENU_ITEMS(const QString &argument, OCC::SocketListe
             listener->sendMessage(QLatin1String("MENU_ITEM:EMAIL_PRIVATE_LINK::") + tr("Send private link by email..."));
         }
     }
+
+    // Placeholder download action
+    if (syncFolder) {
+        auto placeholderSuffix = QStringLiteral(OWNCLOUD_PLACEHOLDER_SUFFIX);
+        bool hasPlaceholderFile = false;
+        for (const auto &file : files) {
```

Comment:
i'd use  std::all_of
You want to show this menu if all files are placeholder, right?
This code will work if only one of the files is a placeholder.

Reply:
I thought it'd be nice to show this as long as any one file in the selection was a placeholder. That way you can select a bunch of folder contents and click - don't have to make sure to skip the odd files that aren't placeholders. We could even have this option say "Download 37 files that are currently placeholders".